internal/convertor/device/device.go

```diff
@@ -92,12 +98,17 @@ func NewDevice(dcimInfo *dcim.NetworkDevice, devicesData *repository.AssetsPerDe
 		return nil, fmt.Errorf("no route-policies found for %s", dcimInfo.Hostname)
 	}
 
+	device.SNMP, ok = devicesData.SNMP[dcimInfo.Hostname]
+	if !ok {
```

Comment:
SNMP being optional, you should not fail here. (same than line 76)

Reply:
this part was refactred